ipatests/test_integration/test_commands.py

```diff
@@ -1646,6 +1646,89 @@ def test_sid_generation(self):
             tasks.kinit_admin(self.master)
             self.master.run_command(['ipa', 'user-del', user])
 
+    def test_sequence_processing_ipaexternalgroup(self):
+        """Test for sequence processing failures
+
+        Issues have been found for group_add sequence processing with
+        server context. This test checks that groups have correct userclass
+        when external is set to true or false with group-add.
+
+        related: https://pagure.io/freeipa/issue/9349
+        """
+        user_code_script = textwrap.dedent("""
+            from ipalib import api, errors
+            api.bootstrap_with_global_options(context='server')
+            api.finalize()
+            if api.env.in_server:
+                api.Backend.ldap2.connect()
+            else:
+                api.Backend.rpcclient.connect()
+            try:
+                api.Command["group_del"]("testgroup1")
+            except errors.NotFound:
+                pass
+            try:
+                api.Command["group_del"]("testgroup2")
+            except errors.NotFound:
+                pass
```

Comment:
I wonder if this cleanup can be moved to a fixture instead, and basically repeat the group deletion before and after the yield for all possible test groups. An example is expire_cert_critical() in test_ipa_cert_fix.py. That or use a list of groups to delete instead and iterate over it in the pushed script. IMHO it should clean up after itself as well.